pkg/sync/v6/sync.go

```diff
@@ -481,6 +497,19 @@ type sync struct {
 	cli        *discovery.DiscoveryClient
 	dyn        dynamic.ClientPool
 	grs        []*discovery.APIGroupResources
+
+	config config
+}
+
+// config defines sync pod behaviour based on conditional logic in the cluster.
+// if set to false - components becomes unmanaged.
+// we need to make sure this for the nested dependencies (example: resource in the namespace)
+// we set these flags right time in the sync process
+type config struct {
```

Comment:
`config` seems like much too generic of a name to me, especially when we're including it in a struct that has a number of other different configs already. Can we give this a more targeted name?

Reply:
reconcileConfig